### PR TITLE
fix: update conditional profiles

### PR DESCRIPTION
**Pull request by ajasnosz**
# Description

Improving conditional profiles by adding option to negate requested operation and adding filtering with regex.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Refactor/improvement
- [x] This change requires a documentation update

## How Has This Been Tested?

manual tests, unit tests, integration tests

## Checklist

- [x] My commit message is [conventional](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] I have run pre-commit on all files before creating the PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: splunk_connect_for_snmp/inventory/tasks.py [hunk 1]
```diff
@@ -284,12 +284,20 @@ def create_profile(profile_name, frequency, varBinds, records):
...
         "equals": "$eq",
         "gt": "$gt",
         "lt": "$lt",
         "in": "$in",
+        "regex": "$regex",
+    }
+
+    negative_profiles_mapping = {
+        "equals": "$ne",
+        "gt": "$lte",
```
Comment (omrozowicz-splunk) [inline splunk_connect_for_snmp/inventory/tasks.py:297]: is it `$lte`? not `$lt`?

Comment (ajasnosz) [reply splunk_connect_for_snmp/inventory/tasks.py:297]: Yes, I followed with the logic as if we used `not` operator on `gt` that would return something less or equal.

Comment (omrozowicz-splunk) [reply splunk_connect_for_snmp/inventory/tasks.py:297]: ok, got it

**Review by omrozowicz-splunk — APPROVED** [review]

**Pull request merged**